dapp/src/components/buySell/ApproveButtonLogic.js

```diff
@@ -0,0 +1,188 @@
+import { ethers } from 'ethers'
+import { fbt } from 'fbt-runtime'
+import { useStoreState } from 'pullstate'
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useState,
+} from 'react'
+import ContractStore from 'stores/ContractStore'
+import analytics from 'utils/analytics'
+import { assetRootPath } from 'utils/image'
+import { useWeb3React } from '@web3-react/core'
+import Media from 'react-media'
+
+// const GLOBAL_MEDIA_QUERIES = {
+//   small: "(max-width: 599px)",
+//   medium: "(min-width: 600px) and (max-width: 1199px)",
+//   large: "(min-width: 1200px)"
+// };
+// const matches = useMedia({ queries: GLOBAL_MEDIA_QUERIES });
+
+const ApproveButtonLogic = forwardRef(
+  (
+    {
+      formHasErrors,
```

Comment:
Any specific reason to using `forwardRef` instead of just passing these values as Component properties?

Reply:
We use an approach to declare functions in the scope of the parent and pass them down to a component like so: https://github.com/OriginProtocol/origin-dollar/blob/master/dapp/src/components/buySell/SwapHomepage.js#L525-L535

The above example functions are all declared inline with the child component, you can also normally declare them on the parent and pass as a reference to child component